catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/ResultIterable.java

```diff
@@ -48,53 +52,117 @@
  */
 public class ResultIterable implements Iterable<Result> {
 
-    private final QueryFunction queryFunction;
+    private QueryFunction queryFunction;
 
-    private final QueryRequest queryRequest;
+    private QueryRequest queryRequest;
+
+    private int maxResultCount;
 
     /**
      * Creates an iterable that will call the {@link CatalogFramework} to retrieve the results
-     * that match the {@link QueryRequest} provided.
+     * that match the {@link QueryRequest} provided. There will be no limit to the number of
+     * results returned.
      *
      * @param catalogFramework reference to the {@link CatalogFramework} to call to retrieve the
      *                         results.
      * @param queryRequest     request used to retrieve the results.
      */
-    public ResultIterable(CatalogFramework catalogFramework, QueryRequest queryRequest) {
-        notNull(catalogFramework, "Catalog framework reference cannot be null");
-        notNull(queryRequest, "Query request cannot be null");
-
-        this.queryFunction = catalogFramework::query;
-        this.queryRequest = queryRequest;
+    public static ResultIterable resultIterable(CatalogFramework catalogFramework,
+            QueryRequest queryRequest) {
+        notNull(catalogFramework, "CatalogFramework cannot be null");
+        return new ResultIterable(catalogFramework, queryRequest, 0);
     }
 
     /**
      * Creates an iterable that will call a {@link QueryFunction} to retrieve the results
-     * that match the {@link QueryRequest} provided.
+     * that match the {@link QueryRequest} provided. There will be no limit to the number of
+     * results returned.
      *
      * @param queryFunction reference to the {@link QueryFunction} to call to retrieve the
      *                      results.
      * @param queryRequest  request used to retrieve the results.
      */
-    public ResultIterable(QueryFunction queryFunction, QueryRequest queryRequest) {
+    public static ResultIterable resultIterable(QueryFunction queryFunction,
```

Comment:
@rzwiefel and I had discussed this last week and I suggested that the factory methods would probably be a bit cleaner than multiple ctors with an init method. It allows us to keep the `final` fields (not the biggest deal in the world, of course) and sets us up for any future customizations either through more factory methods or by adding a Builder. It was a merely a preference of mine and certainly isn't necessary; if the impact to other work is going to be too large we definitely don't have to make it.